swarm/src/lib.rs

```diff
@@ -1508,38 +1510,44 @@ impl error::Error for DialError {
     }
 }
 
-/// Dummy implementation of [`NetworkBehaviour`] that doesn't do anything.
-#[derive(Clone)]
-pub struct DummyBehaviour {
-    keep_alive: KeepAlive,
-}
+/// Implementation of [`NetworkBehaviour`] that doesn't do anything other than keep all connections alive.
+pub struct KeepAliveBehaviour;
+
+impl NetworkBehaviour for KeepAliveBehaviour {
+    type ConnectionHandler = KeepAliveConnectionHandler;
+    type OutEvent = void::Void;
 
-impl DummyBehaviour {
-    pub fn with_keep_alive(keep_alive: KeepAlive) -> Self {
-        Self { keep_alive }
+    fn new_handler(&mut self) -> Self::ConnectionHandler {
+        KeepAliveConnectionHandler
     }
 
-    pub fn keep_alive_mut(&mut self) -> &mut KeepAlive {
-        &mut self.keep_alive
+    fn inject_event(
+        &mut self,
+        _: PeerId,
+        _: ConnectionId,
+        event: <Self::ConnectionHandler as ConnectionHandler>::OutEvent,
+    ) {
+        void::unreachable(event)
     }
-}
 
-impl Default for DummyBehaviour {
-    fn default() -> Self {
-        Self {
-            keep_alive: KeepAlive::No,
-        }
+    fn poll(
+        &mut self,
+        _: &mut Context<'_>,
+        _: &mut impl PollParameters,
+    ) -> Poll<NetworkBehaviourAction<Self::OutEvent, Self::ConnectionHandler>> {
+        Poll::Pending
     }
 }
 
-impl NetworkBehaviour for DummyBehaviour {
-    type ConnectionHandler = handler::DummyConnectionHandler;
+/// Implementation of [`NetworkBehaviour`] that doesn't do anything other than keep all connections alive.
+pub struct DummyNetworkBehaviour;
```

Comment:
Our naming conventions aren't crystal clear here :)

There is somewhat of a naming clash between `behaviour::KeepAlive` and the actual `KeepAlive` enum now. I think we can rename the latter one to `KeepAlivePolicy` to make that clearer. This would also allow us to rename the `KeepAliveConnectionHandler` to just `KeepAlive` and have the module name differentiate between the two.

Reply:
Good point on the clash between `KeepAlive` _policy_ and _behaviour_.

I don't feel strongly about this. I tend towards keeping the names as is. `KeepAlive` is widely used, the `KeepAliveBehaviour` is only used in test code. Thus I don't think this justifies a breaking change.

Reply:
This is resolved with the latest rename.